### ISSUE TITLE
CUPS problem, where is the package?

### ISSUE THREAD

**Issue by davigre** [Entware-ng/Entware-ng#80]
I want to change the default printer server ({p910nd} /usr/sbin/p9100d -b -f /dev/usb/lp0 0) to CUPS server so then I can use it as a "google cloud printer". I have a SPLIX format printer (Samsung ML1740) so I install the following packages:

- cups
- cups-client (to get lp command)
- splix

"p910nd' works great as raw printer (I need printer driver in Windows).

I put down "p910nd" daemon so it frees /dev/usb/lp0 port that is used by default CUPS Printer named "USB". Install the (new) printer driver with the already working win7 drivers but redirect it to the CUPS "USB" printer ( http://192.168.1.1:631/printers/USB ) expecting the same results as using "p910nd", but    when sending a test page the state of the job in CUPS say': "Waiting for printer to become available." and is never printed.

I don't know what to do next. ideas? Where can I get CUPS package, I can't find it?

Using "p910nd" server and CUPS with a new created printer I can print in raw format, but I don't want to use 2 daemons. Even using 2 daemons I can't print a pdf file with ('lp test.pdf') command, it sends "lp: Unsupported document-format "application/pdf"" message.

**Comment by ryzhovau** [comment]
> Even using 2 daemons I can't print a pdf file with ('lp test.pdf') command, it sends "lp: Unsupported document-format "application/pdf"" message.

It's means no pdf filter for current printer driver. I've saw how some other users printed pdf files via `pdftops` utility from `xpdf` package.

**Comment by davigre** [comment]
Thank you. You give me a clue...

My (non PS printer) Samsung ML1740 uses a format known as 'qpdl'. The SPLIX package install 'pstoqpdl' and 'rastertoqpdl' on the CUPS 'filter' path (/opt/lib/cups/filter). I start trying to print a ps file first. I found 1 problem:

'pstoraster' filter don't exists. Even 'pstoqpdl' utility need it (I don't know why) as shown in log CUPS file.

I only need a PS to CUPS Raster format filter ('pstoraster'), but none CUPS, xpdf, ghostscript have it. This utility is meant to be in ghostscript package. But I found that has been taken off the ghostscript package to a package know as 'cups-filters.

But I think that will help to make compile CUPS PDF Filter as shown in 
[PDF as Standard Print Job Format](http://www.linuxfoundation.org/collaborate/workgroups/openprinting/pdf_as_standard_print_job_format)

thank you.
